### PR TITLE
add Microsoft.DevOpsInfrastructure as allowed services for subnet delegation

### PR DESCRIPTION
**Pull request by schdief**
https://learn.microsoft.com/en-us/azure/devops/managed-devops-pools/prerequisites?view=azure-devops&tabs=azure-portal#azure-portal-1

## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

Microsoft released a new service called Managed DevOps agents. To use it, one needs to delegate a subnet to the service Microsoft.DevOpsInfrastructure according to the docs: https://learn.microsoft.com/en-us/azure/devops/managed-devops-pools/prerequisites?view=azure-devops&tabs=azure-portal#azure-portal-1

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_subnet` - support `Microsoft.DevOpsInfrastructure` as delegation service

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/network/subnet_resource.go [hunk 1]
```diff
@@ -59,6 +59,7 @@ var subnetDelegationServiceNames = []string{
 	"Microsoft.DBforPostgreSQL/singleServers",
 	"Microsoft.DelegatedNetwork/controller",
 	"Microsoft.DevCenter/networkConnection",
+	"Microsoft.DevOpsInfrastructure",
```
Comment (ajaykn) [inline internal/services/network/subnet_resource.go:62]:
```suggestion
	"Microsoft.DevOpsInfrastructure/pools",
```

same in another change as well

Comment (schdief) [reply internal/services/network/subnet_resource.go:62]: thanks :)

**Comment by haflidif** [comment]
Resolves: #26412

**Comment by schdief** [comment]
> Resolves: #26412

@haflidif no it doesn't, this PR merely allows to delegate the subnet, but it doesn't provide a new module to deploy the Managed DevOps Pool itself, please remove the reference so the issue isn't closed with the merge

**Comment by haflidif** [comment]
> > Resolves: #26412
> 
> @haflidif no it doesn't, this PR merely allows to delegate the subnet, but it doesn't provide a new module to deploy the Managed DevOps Pool itself, please remove the reference so the issue isn't closed with the merge

My apologizes for misreading the Issue, I was to eager to link these two together, thought the issue was mainly about adding the Delegation on the Subnet.

It's been removed now.

**Comment by schdief** [comment]
> > > Resolves: #26412
> > 
> > 
> > @haflidif no it doesn't, this PR merely allows to delegate the subnet, but it doesn't provide a new module to deploy the Managed DevOps Pool itself, please remove the reference so the issue isn't closed with the merge
> 
> My apologizes for misreading the Issue, I was to eager to link these two together, thought the issue was mainly about adding the Delegation on the Subnet.
> 
> After reading it better now, I see it's a request for the whole resource, not only the requirements it needs on the Subnet.
> 
> It's been removed now, thanks for paying attention to the details.

no worries, thanks for trying to help :)

**Comment by wjdavis5** [comment]
very happy someone beat me to this ;-) 
Thanks!

**Review by katbyte — APPROVED** [review]
LGTM ☄️

**Pull request merged**